openbb_platform/core/openbb_core/app/extension_loader.py

```diff
@@ -0,0 +1,152 @@
+"""Extension Loader."""
+
+from enum import Enum
+from functools import lru_cache
+from typing import Any, Dict, Optional
+
+from importlib_metadata import EntryPoint, EntryPoints, entry_points
+
+from openbb_core.app.model.abstract.singleton import SingletonMeta
+from openbb_core.app.model.extension import Extension
+
+
+class OpenBBGroups(Enum):
+    """OpenBB Extension Groups."""
+
+    core = "openbb_core_extension"
+    provider = "openbb_provider_extension"
+    obbject = "openbb_obbject_extension"
+
+
+class ExtensionLoader(metaclass=SingletonMeta):
+    """Extension loader class."""
+
+    def __init__(
+        self,
+    ) -> None:
+        """Initialize the extension loader."""
+        self._obbject_entry_points: EntryPoints = self._sorted_entry_points(
+            group=OpenBBGroups.obbject.value
+        )
+        self._core_entry_points: EntryPoints = self._sorted_entry_points(
+            group=OpenBBGroups.core.value
+        )
+        self._provider_entry_points: EntryPoints = self._sorted_entry_points(
+            group=OpenBBGroups.provider.value
+        )
+        self._obbject_objects: Dict[str, Extension] = {}
+        self._core_objects: Dict[str, Any] = {}
```

Comment:
Would be very cool to not lose type hinting here (and in the below reference to this variable) but I couldn't make the forward reference work.
This is bc I need to import such classes (Provider and Router) inside the `load_core` and `load_provider` methods to avoid circular imports.
It's not a showstopper but, any ideas @montezdesousa?

Reply:
Did you try this?

```python
if TYPE_CHECKING:
    from openbb_core.app.router import Router
    from openbb_core.provider.abstract.provider import Provider

...

        self._core_objects: Dict[str, Router] = {}
        self._provider_objects: Dict[str, Provider] = {}
```

Reply:
perfect 📞 call